vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java

```diff
@@ -126,6 +126,31 @@ public WebElement assertIfElementExists(String businessDescription, Locator loca
         return assertIfElementExists(businessDescription, uiContext.getSearchContext(), locator);
     }
 
+    @Override
+    public Optional<WebElement> assertElementExists(String description, Locator locator)
+    {
+        List<WebElement> elements = searchActions.findElements(uiContext.getSearchContext(), locator);
+        if (elements.size() > 1)
+        {
+            String assertionFormat = "The number of elements found by the locator %s is %d, but expected 1";
+            softAssert.recordFailedAssertion(String.format(assertionFormat, locator, elements.size()));
+        }
+        else if (doesOnlyOneElementExist(description, locator, elements))
+        {
+            return Optional.of(elements.get(0));
+        }
+        return Optional.empty();
+    }
+
+    private boolean doesOnlyOneElementExist(String description, Locator locator, List<WebElement> elements)
+    {
+        boolean passed = elements.size() == 1;
+        String condition = passed ? " is" : " is not";
+        String assertionMessage = description + condition + " found by the locator " + locator;
+        return passed ? softAssert.recordPassedAssertion(assertionMessage)
+                : softAssert.recordFailedAssertion(assertionMessage);
```

Comment:
```java
boolean passed = elements.size() == 1;
StringBuilder assertionMessageBuilder = new StringBuilder(description).append(" is ");
if (!passed)
{
    assertionMessageBuilder.append("not ");
}
assertionMessageBuilder.append("found by the locator ").append(locator);
String assertionMessage = assertionMessageBuilder.toString();
return passed ? softAssert.recordPassedAssertion(assertionMessage)
        : softAssert.recordFailedAssertion(assertionMessage);
```

Reply:
hmm what is the advantage of this code snippet? btw some time ago we decided to gradually get rid of string builders due to jep 280

Reply:
> hmm what is the advantage of this code snippet? 

no more duplicated ` is `

> btw some time ago we decided to gradually get rid of string builders due to jep 280

that was about old rule that each string concatenation should be done using `StringBuilder`, but in this case `StringBuilder` is used to build a string, you can try to re-write using `+`, but it won't improve readability

Reply:
imho concise code is prevalent over duplication of some small part of text